crates/metadata-providers/src/objstore/optimistic_store.rs

```diff
@@ -46,14 +50,39 @@ enum OnDiskValue<'a> {
     V1(Cow<'a, VersionedValue>, u64),
 }
 
-fn tagged_value_to_versioned_value(tagged_value: &TaggedValue) -> anyhow::Result<VersionedValue> {
-    let on_disk: OnDiskValue<'static> = ciborium::from_reader(tagged_value.bytes.as_ref())?;
-    match on_disk {
-        OnDiskValue::V1(cow, _) => Ok(cow.into_owned()),
-    }
+// similar to OnDiskValue, but without serde support.
+// and will be used for bilrost encoding.
+#[derive(Debug, Clone, bilrost::Message)]
+struct SaltedVersionedValue {
+    #[bilrost(1)]
+    salt: u64,
+    #[bilrost(2)]
+    value: VersionedValue,
+}
+
+fn tagged_value_to_versioned_value(
+    tagged_value: TaggedValue,
+) -> anyhow::Result<(Tag, VersionedValue)> {
+    let tag = tagged_value.tag;
+    let versioned_value = match tagged_value.content.encoding {
+        ValueEncoding::Cbor => {
+            let on_disk: OnDiskValue<'static> =
+                ciborium::from_reader(tagged_value.content.bytes.as_ref())?;
+            match on_disk {
+                OnDiskValue::V1(cow, _) => Ok(cow.into_owned()),
+            }
+        }
+        ValueEncoding::Bilrost => SaltedVersionedValue::decode(tagged_value.content.bytes.as_ref())
+            .context("failed to decode bilrost")
+            .map(|v| v.value),
+    }?;
+    Ok((tag, versioned_value))
 }
 
 impl OptimisticLockingMetadataStore {
+    // TODO: Switch to bilrost in 1.6.0
+    const DEFAULT_ENCODING: ValueEncoding = ValueEncoding::Cbor;
```

Comment:
This is redundant, ValueEncoding implements `Default` and it'll be only confusing to have two places with default value set.